CHANGELOG.md

```diff
@@ -48,6 +48,8 @@ balances or a single balance by denom when the `denom` query parameter is presen
 * [\#5785](https://github.com/cosmos/cosmos-sdk/issues/5785) JSON strings coerced to valid UTF-8 bytes at JSON marshalling time
 are now replaced by human-readable expressions. This change can potentially break compatibility with all those client side tools
 that parse log messages.
+* (codec) [\#5799](https://github.com/cosmos/cosmos-sdk/pull/5799) Now we favor the use of MarshalBinaryBare instead of LengthPrefixed in all cases that are not needed,
```

Comment:
@jgimeno can we instead state _which_ commands break here. AFAIK, it's only `tx encode/decode`.

Reply:
This is a bit concerning: https://github.com/cosmos/gaia/commit/680bb196d6c5ece0c78875dff0175fdd780711ac#diff-ef04ec516e080b802253864b259addb3L621

Because of the changes in coins serialization, I'd advise to warn the users about the impact of the change instead of mentioning every single affected command (which could potentially be many)

Reply:
Plus, @jgimeno I'd advise to revert the changes in keys serialization. That risks to be a very painful issue for the users

Reply:
Errr what you linked @alessio has nothing to do with serialization. That's a completely orthogonal change and has been stated in the CHANGELOG already.

Reply:
https://github.com/cosmos/cosmos-sdk/pull/5799